### PR TITLE
Clash 1.9 update

### PR DESCRIPTION
**Pull request by kleinreact**
This PR bumps Clash to the latest required 1.9 version (fixing the `xpmCdcHandshake` primitive)  and applies the corresponding interface changes. In particular, this includes

* the correct usage of `clockWizardDifferential` getting rid of the _improvised reset syncers_,
* renaming of clock domains from `BasicX` to `ExtX` for all clocks that originate from a differential input, and
* changing the reset kind of all `ExtX` domains to `Asynchronous` resets.

---

_Note that `clockControlDemo0` currently does not compile due to a Clash bug introduced since the previously used version, which is under investigation right now._

### CODE REVIEW

**Review by DigitalBrains1 — COMMENTED** [review]
I think the difference between an improvised reset synchroniser with `xpmCdcSingle` and the bog-standard `resetSynchronizer` that is part of the safe PLL functions is that the former includes a `max_delay` constraint and an `ASYNC_REG` attribute whereas the latter includes none of those things. So I think the improvisation is qualitatively better...

**Comment by leonschoorl** [comment]
https://github.com/clash-lang/clash-compiler/pull/2624 should allow clockControlDemo0 to build again

**Review by martijnbastiaan — APPROVED** [review]
@kleinreact Could you check whether @leonschoorl's patch works?

**Comment by kleinreact** [comment]
> @kleinreact Could you check whether @leonschoorl's patch works?

Negative :disappointed:. @leonschoorl: Is that new error related to your fixes?

**Comment by martijnbastiaan** [comment]
The output is confusing: it prints a warning (which we should be able to ignore if it doesn't come back later as an error). The real error is:

```
    Clash.Cores.Xilinx.DcFifo.Internal.BlackBoxes.dcFifoTF: dcFifo only supports synchronous resets
```

Do we connect dc fifo to an async reset? Because that would be an issue yeah...

**Comment by kleinreact** [comment]
> Do we connect dc fifo to an async reset?

The only related change is the update of the external clock domains to use asynchronous resets, which is required for the safe variants of clock wizards and looks to me like the right way of doing it? So yes, if that's correct then we are connecting the dc fifo to an asynchronous reset somewhere in that instance.

**Comment by martijnbastiaan** [comment]
Hmmyeah, I was typing a response in another PR:

Screenshot from 2024-01-02 14-29-26

So in short, we don't have a good way of signalling "this is a signal that's unrelated to any (visible) domain".

**Comment by kleinreact** [comment]
But it still means we are using a "free reset" with dcFifo, which is unintended behavior, right? So the error indication points to a true error then?

**Comment by martijnbastiaan** [comment]
> But it still means we are using a "free reset" with dcFifo, which is unintended behavior, right? So the error indication points to a true error then?

Yes, it is unintended behavior. The reset should be synchronized. The error just crops up for the "wrong" reasons.

File: bittide-instances/src/Bittide/Instances/Domains.hs [hunk 1]
```diff
@@ -14,14 +14,17 @@ import Bittide.Arithmetic.Ppm
 import Data.Proxy
 
 createDomain vXilinxSystem{vName="Basic125", vPeriod= hzToPeriod 125e6}
+createDomain vXilinxSystem{vName="Ext125", vPeriod= hzToPeriod 125e6, vResetKind=Asynchronous}
```
[the remark this answers is not in the document]
Comment (lmbollen) [reply bittide-instances/src/Bittide/Instances/Domains.hs:17]: So why the different records?

**Pull request merged**